packages/web-components/src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,204 @@
+import { attr, FASTElement, nullableNumberConverter } from '@microsoft/fast-element';
+import { uniqueId } from '@microsoft/fast-web-utilities';
+
+const SUPPORTS_ANCHOR_POSITIONING = CSS.supports('anchor-name: --a');
+
+/**
+ * A Tooltip Custom HTML Element.
+ * Based on ARIA APG Tooltip Pattern {@link https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/ }
+ * @public
+ */
+export class Tooltip extends FASTElement {
+  /**
+   * The attached element internals
+   */
+  public elementInternals = this.attachInternals();
+
+  /**
+   * The item ID
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: id
+   */
+  @attr
+  public id: string = uniqueId('tooltip-');
+
+  /**
+   * Set the delay for the tooltip
+   */
+  @attr({ converter: nullableNumberConverter })
+  public delay?: number;
+
+  /**
+   * The default delay for the tooltip
+   * @internal
+   */
+  private defaultDelay: number = 250;
+
+  /**
+   * Set the positioning of the tooltip
+   */
+  @attr
+  public positioning?: string;
```

Comment:
Is this a string or a constant?

Reply:
It has a proper type now. Please review.